### PR TITLE
fix #27

### LINKED ISSUE #27: package.json#repository is not a string but an object
The current implementation fails on getRepo(baton), since the repository field in the package is not a string but an object. Instead it needs to do the replacement on the repository#url.

Also, the regex used for the replacement is far too narrow (github only) should be replaced by something different.

### PR DESCRIPTION
**Pull request by silkentrance**
- support repository object instead of just string
- upgrade to latest gh-got
- fix test expectations

### CODE REVIEW

**Comment by silkentrance** [comment]
@ngryman some test fail due to github API limits. Other tests fail due to chdir(__dirname) not being called due to failing test.

See https://github.com/sindresorhus/gh-got#token for information on how to fix this. But this will require you to get a github API token...

File: .gitignore [hunk 1]
```diff
@@ -3,6 +3,7 @@
 .todo
 coverage/
 dist/
+package-lock.json
```
Comment (ngryman) [inline .gitignore]: I'm ok with adding a lock file, it's 2018 😄

File: index.js [hunk 2]
```diff
@@ -47,8 +51,8 @@ function update(baton) {
       replace({
         files: path.resolve(baton.dir, filename),
         // eslint-disable-next-line
-        replace: /\[\/\/\]: contributor-faces(?:(?:\n.*)+\[\/\/\]: contributor-faces)?/,
-        with: `[//]: contributor-faces\n${baton.html}\n[//]: contributor-faces`
+        from: /\[\/\/\]: contributor-faces(?:(?:\n.*)+\[\/\/\]: contributor-faces)?/,
+        to: `[//]: contributor-faces\n${baton.html}\n[//]: contributor-faces`
```
Comment (ngryman) [inline index.js]: Just to make sure, `from` and `to` are aliases to `replace` and `with`?

File: .gitignore (hunk 1 above)
Comment (silkentrance) [reply .gitignore]: Ups, this shouldn't have been in there. I added a .gitignore entry for this.

File: index.js (hunk 2 above)
Comment (silkentrance) [reply index.js]: replace and with have been deprecated and it is advised to use their replacements from and to instead.

File: .gitignore (hunk 1 above)
Comment (silkentrance) [reply .gitignore]: package-lock.json is gone.

Comment (ngryman) [reply .gitignore]: Oh I meant that I was ok with adding the `package-lock.json` actually. But that's ok!

**Comment by ngryman** [comment]
Published as `1.0.2` 🎉

**Pull request merged**